blockly-core/core/ui/contract_editor/example_view.js

```diff
@@ -52,14 +50,19 @@ Blockly.ExampleView.prototype.initializeTestButton = function (buttonText, iconC
 };
 
 Blockly.ExampleView.prototype.testExample_ = function () {
-  this.resultText.innerHTML = this.onExampleRun_(this.block_);
+  this.contractEditor_.resetExampleViews();
```

Comment:
I'm not a huge fan of passing a bunch of callbacks to a constructor, as I end up finding this hard to follow.

Another approach we could take would be to make example_view an event emitter. When it emits "testing-example" the contract editor clears all example view and when it emits "reset" we would ask it for it's example block and reset that block.